llm-complete-guide/steps/eval_retrieval.py

```diff
@@ -94,6 +95,9 @@ def query_similar_docs(
     vector_store_name = find_vectorstore_name()
     if vector_store_name == "pgvector":
         conn = get_db_conn()
+    elif vector_store_name == "pinecone":
+        # in pipeline runs, always use staging index
+        pinecone_index = get_pinecone_client(model_version_name_or_id="staging")
```

Comment:
WHy is this hardcoded? this will fail for people starting from scratch ?

Reply:
@wjayesh

Reply:
Shouldn't fail for people starting from scratch. Because we don't check for the staging tag anywhere. This could be anything other than production. We only do the check for production

Reply:
Yes, I assumed that all the development workflow involving experimentation and evals runs on non prod versions. Once you figure out that it works and you move to production you just need the rag pipeline to build the prod index. The eval pipeline is redundant